### PR TITLE
AVX-16378: check active-active for transit ha enabled unmapped policy based conn

### PR DESCRIPTION
**Pull request by shuoliugit**

### CODE REVIEW

File: aviatrix/resource_aviatrix_site2cloud.go [hunk 1]
```diff
@@ -439,16 +439,32 @@ func resourceAviatrixSite2CloudCreate(d *schema.ResourceData, meta interface{})
...
 
+	if s2c.ConnType != "mapped" && s2c.ConnType != "unmapped" {
+		return fmt.Errorf("'connection_type' should be 'mapped' or 'unmapped'")
+	}
+
+	gateway := &goaviatrix.Gateway{
+		GwName: s2c.GwName,
+	}
+
+	gw, err := client.GetGateway(gateway)
```
Comment (zhiqiangjin) [inline aviatrix/resource_aviatrix_site2cloud.go:450]: We should avoid calling more APIs here.

File: aviatrix/resource_aviatrix_site2cloud.go [hunk 2]
```diff
@@ -439,16 +439,32 @@ func resourceAviatrixSite2CloudCreate(d *schema.ResourceData, meta interface{})
...
+	if err != nil {
+		return fmt.Errorf("couldn't find Aviatrix Gateway %s: %v", s2c.GwName, err)
+	}
+
+	if gw.IsTransit == "yes" {
```
Comment (zhiqiangjin) [inline aviatrix/resource_aviatrix_site2cloud.go]: If IsTransit has to be added, it should be a boolean instead of a string.

**Review by zhiqiangjin — CHANGES_REQUESTED** [review]

File: aviatrix/resource_aviatrix_site2cloud.go (hunk 1 above)
Comment (shuoliugit) [reply aviatrix/resource_aviatrix_site2cloud.go:450]: this is a special case only for transit gateway, any suggestions about how to get the type of a gateway?

File: aviatrix/resource_aviatrix_site2cloud.go (hunk 2 above)
Comment (shuoliugit) [reply aviatrix/resource_aviatrix_site2cloud.go]: changed IsTransit to TransitVpc

File: aviatrix/resource_aviatrix_site2cloud.go (hunk 1 above)
Comment (shuoliugit) [reply aviatrix/resource_aviatrix_site2cloud.go:450]: have to call an API here to get the gateway type

File: aviatrix/resource_aviatrix_site2cloud.go [hunk 3]
```diff
@@ -439,16 +439,32 @@ func resourceAviatrixSite2CloudCreate(d *schema.ResourceData, meta interface{})
...
+		return fmt.Errorf("'connection_type' should be 'mapped' or 'unmapped'")
+	}
+
+	gateway := &goaviatrix.Gateway{
+		GwName: s2c.GwName,
+	}
+
+	gw, err := client.GetGateway(gateway)
+	if err != nil {
+		return fmt.Errorf("couldn't find Aviatrix Gateway %s: %v", s2c.GwName, err)
```
Comment (zhiqiangjin) [inline aviatrix/resource_aviatrix_site2cloud.go]: This error message will be no good if err == ErrNotFound

**Review by zhiqiangjin — CHANGES_REQUESTED** [review]

**Review by mlin-aviatrix — APPROVED** [review]

**Review by zhiqiangjin — APPROVED** [review]

**Pull request merged**